### ISSUE TITLE
Unsaved files' state is rolling back after tab switching

### ISSUE THREAD

**Issue by Shtokins** [microsoft/vscode#134221]
I have many opened files per time. Often it's the same files in different windows. Last two days I several times noticed a situation when changes i just made had disappeared. I make changes, switch to another tab, then switch back - and I should do them again - all code returns in several minutes in the past.
I noticed it for different files and different projects.
It's hard to reproduce it intentionally

**Comment by bpasero** [comment]
We do load the contents of the file from disk anytime you open a tab, but that would not happen when the tab is dirty (i.e. we never replace changes that have not been saved to disk).

Do you have auto save enabled or saved the tab before switching?

**Comment by Shtokins** [comment]
No auto save
Tab wasn't saved (this one is not clear. I think in some cases it could be even saved with new code before old one appears again)

Of course I know how it works ) I have used VSCode for 4 years and first and second time I noticed this issue I was thinking that I did something wrong. Press and hold ctrl+z or I don't know what. But it repeated more than 5 times per 2 days in different files/folders/projects/stack. Yesterday I started save every short changes and even commit them to GIT in order to prevent this thing's repeating...
